From 724b6f682badba8db06168fe7d9a406098962ff1 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Thu, 24 Dec 2015 14:54:48 +0530
Subject: [PATCH 10/14] USBC/DPM: Set phy swap state to handle power role swap
 events.

DPM should set the swap state to phy when power role change to
POWER_ROLE_SWAP so that phy can handle the events during
pr_swap and clear once the role changes to sink/src/none.

Change-Id: I8cbced6b77d89832615c23c5c843723853bbe5d2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15496
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/452124
---
 drivers/usb/typec/pd/devpolicy_mgr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/typec/pd/devpolicy_mgr.c b/drivers/usb/typec/pd/devpolicy_mgr.c
index e8d4f72..1fd7652 100644
--- a/drivers/usb/typec/pd/devpolicy_mgr.c
+++ b/drivers/usb/typec/pd/devpolicy_mgr.c
@@ -519,6 +519,7 @@ static void dpm_update_power_role(struct devpolicy_mgr *dpm,
 					__func__);
 			goto update_prole_err;
 		}
+		typec_set_swap_state(dpm->phy, true);
 		break;
 	case POWER_ROLE_SINK:
 		dpm->consumer_state = CABLE_ATTACHED;
@@ -548,6 +549,8 @@ static void dpm_update_power_role(struct devpolicy_mgr *dpm,
 		pr_debug("DPM:%s: unknown pwr role!!\n", __func__);
 		goto update_prole_err;
 	}
+	if (dpm->prev_prole == POWER_ROLE_SWAP)
+		typec_set_swap_state(dpm->phy, false);
 	dpm->prev_prole = cur_prole;
 	dpm->cur_prole = prole;
 
-- 
1.9.1

